File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/SavedAddress/GetSavedAddressUseCase.java
package com.lattis.ellipse.domain.interactor.SavedAddress;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.SavedAddress;
import com.lattis.ellipse.domain.repository.SavedAddressRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

public class GetSavedAddressUseCase extends UseCase<List<SavedAddress>>

{

    private SavedAddressRepository savedAddressRepository;
    private SavedAddress savedAddress;

    @Inject
    public GetSavedAddressUseCase(ThreadExecutor threadExecutor,
                               PostExecutionThread postExecutionThread,
                               SavedAddressRepository savedAddressRepository) {
        super(threadExecutor, postExecutionThread);
        this.savedAddressRepository = savedAddressRepository;
    }


    @Override
    protected Observable<List<SavedAddress>> buildUseCaseObservable() {
        return this.savedAddressRepository.getSavedAddresses();
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/saveaddress/GetSavedAddressUseCase.kt
package com.lattis.domain.usecase.saveaddress

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.SavedAddressRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.SavedAddress
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetSavedAddressUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val savedAddressRepository: SavedAddressRepository
) : UseCase<List<SavedAddress>>(threadExecutor, postExecutionThread) {
    private val savedAddress: SavedAddress? = null
    override fun buildUseCaseObservable(): Observable<List<SavedAddress>> {
        return savedAddressRepository.savedAddresses!!
    }

}